features/FEATURE_LWIP/TESTS/mbedmicro-net/udp_echo_parallel/main.cpp

```diff
@@ -123,9 +123,20 @@ class Echo {
                 iomutex.lock();
                 printf("[ID:%01d][%02d] success #%d\n", id, i, success);
                 iomutex.unlock();
+
+                i++;
+                continue;
             }
 
-            i++;
```

Comment:
You should keep this `i` here. It is keeping track of the total number of loops, not the number of successes.